src/subdomains/generic/user/models/bank-data/bank-data.service.ts

```diff
@@ -17,6 +21,62 @@ export class BankDataService {
     private readonly specialAccountService: SpecialExternalAccountService,
   ) {}
 
+  @Cron(CronExpression.EVERY_MINUTE)
+  @Lock(1800)
+  async checkAndSetActive() {
+    if (DisabledProcess(Process.BANK_DATA_VERIFICATION)) return;
+    await this.checkUnverifiedBankDatas();
+  }
+
+  async checkUnverifiedBankDatas(): Promise<void> {
+    const entities = await this.bankDataRepo.find({
+      where: { active: Not(true), type: Not(In([BankDataType.IDENT, BankDataType.USER])), comment: IsNull() },
+      relations: { userData: true },
+    });
+
+    for (const entity of entities) {
+      await this.verifyBankData(entity);
+    }
+  }
+
+  async verifyBankData(entity: BankData): Promise<void> {
+    const existing = await this.bankDataRepo.findOne({
+      where: { iban: entity.iban, active: true },
+      relations: { userData: true },
+    });
+
+    const errors = this.getBankDataVerificationErrors(entity, existing);
+
+    if (errors.length === 0) {
+      if (!existing) {
+        await this.bankDataRepo.update(...entity.activate());
+      } else {
+        const existingError = [...(existing.comment?.split(';') ?? []), BankDataVerificationError.NEW_BANK_IN_ACTIVE];
+
+        await this.bankDataRepo.update(...existing.deactivate(existingError.join(';')));
```

Comment:
Hm don't know if that's a good idea 🤔 I like the idea to have a kind of history to every bankData. With that I would say you can can analyze errors or something like that much easier because you see what's happened there

Reply:
Ok 😉